### PR TITLE
Create phx.gen.ex

### PR DESCRIPTION
**Pull request by BarbaricCorgi**
This task describe in detail what every generator do. specially the differences between the resource related tasks

### CODE REVIEW

File: lib/mix/tasks/phx.gen.ex [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    phx.gen.auth : Generates authentication logic for a resource.
+    phx.gen.presence : Generates a Presence tracker
+    phx.gen.channel : Generates a Phoenix channel and its corresponding tests
+    phx.gen.notifier : Generates a notifier that delivers emails by default
+    phx.gen.socket : Generates a Phoenix socket handler
+
+    Static generators:
+    phx.gen.cert : Generates a self-signed certificate for HTTPS testing
+    phx.gen.secret : Generates a random key 64 characters long
+    "
```
Comment (josevalim) [inline lib/mix/tasks/phx.gen.ex:48]:
```suggestion
    Mix.Task.run("help", ["--search", "phx.gen."])
```

Comment (josevalim) [reply lib/mix/tasks/phx.gen.ex:48]: The above should achieve a similar result, without requiring us to remember to keep the task up to date. WDYT?

File: lib/mix/tasks/phx.gen.ex [hunk 2]
```diff
@@ -0,0 +1,50 @@
+defmodule Mix.Tasks.Phx.Gen do
+  use Mix.Task
+
+  @shortdoc "Explains in detail what every phx.gen task do"
```
Comment (josevalim) [inline lib/mix/tasks/phx.gen.ex:4]:
```suggestion
  @shortdoc "Lists all available Phoenix generators"
```

File: lib/mix/tasks/phx.gen.ex [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    Other generators:
+    - `phx.gen.auth` : Generates authentication logic for a resource.
+    - `phx.gen.presence` : Generates a Presence tracker
+    - `phx.gen.channel` : Generates a Phoenix channel and its corresponding tests
+    - `phx.gen.notifier` : Generates a notifier that delivers emails by default
+    - `phx.gen.socket` : Generates a Phoenix socket handler
+
+    Static generators:
+    - `phx.gen.cert` : Generates a self-signed certificate for HTTPS testing
+    - `phx.gen.secret` : Generates a random key 64 characters long
```
Comment (josevalim) [inline lib/mix/tasks/phx.gen.ex:26]:
```suggestion
  Lists all available Phoenix generators.
```

File: lib/mix/tasks/phx.gen.ex (hunk 1 above)
Comment (BarbaricCorgi) [reply lib/mix/tasks/phx.gen.ex:48]: yes and we will keep the tasks updated. great choice

File: lib/mix/tasks/phx.gen.ex (hunk 2 above)
Comment (BarbaricCorgi) [reply lib/mix/tasks/phx.gen.ex:4]: I use hexdocs a lot and considering the mix tasks help lists all the commands alphabetically, I had to read every generator one by one to realize how many of them do the same as the previous one plus a little bit more (which is awesome)

so i though, i should improve the documentation (in fact, now i think i should use a markdown table showing the relationship between the CRUD related tasks) and when I went for it I realized I needed to modify Phoenix tasks to achieve it. and considering this is a cross-cutting bit of information, I ended up suggesting this tasks.

In that sense, your idea of automating the listing of tasks is great, I would only add a table describing the CRUD related tasks by order of complexity. will commit them in a minute.

**Review by BarbaricCorgi — COMMENTED** [review]
@josevalim please, may you check this change? do you think documentation may improve this way

this is my first contribution to the project, so your guidance is very enlightening for me

**Comment by josevalim** [comment]
@BarbaricCorgi that's a great idea, i just pruned the table a little bit :)

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**